src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -456,21 +456,17 @@ void Terminal::_WriteBuffer(const std::wstring_view& stringView)
         {
             // TODO: MSFT 21006766
             // This is not great but I need it demoable. Fix by making a buffer stream writer.
-            if (wch >= 0xD800 && wch <= 0xDFFF)
-            {
-                const OutputCellIterator it{ stringView.substr(i, 2), _buffer->GetCurrentAttributes() };
-                const auto end = _buffer->Write(it);
-                const auto cellDistance = end.GetCellDistance(it);
-                i += cellDistance - 1;

```

Comment:
This line causes the loop variable `i` to be decremented if `cellDistance == 0`, which happens to be the case with the HTML from #4145.
Additionally it probably shouldn't have used the `cellDistance` anyways but rather simply increment `i` by 1, whenever 2 code units are extracted from the `stringView`.

Reply:
Why was `cellDistance == 0`? What was the specific code point that caused the problem?

Reply:
I tried to describe that here: https://github.com/microsoft/terminal/pull/4150#issuecomment-572220629
tl;dr: In an iteration with a surrogate pair `GetCellDistance()` returns 1 (as described below), which causes `i` to be "misaligned" within `stringView`. In the next iteration `TextBuffer::Write` will now encounter a low surrogate pair and either advance the iterator again only by 1 character or fail to compose the character and return 0.

Reply:
You can see an example of this misalignment quite well in the picture in this comment: https://github.com/microsoft/terminal/pull/4150#issuecomment-572239776
Instead of it printing 3 glyphs it prints 6, due to `stringView` contains 6 code units and `GetCellDistance()` returning 1 at most. (...and never 2!)

Reply:
Oh no! I accidentially deleted one of my comments here. 😱
Do you have emails from GitHub enabled @miniksa? In that case you should find it there. 😣

Reply:
I do have e-mails from GitHub enabled, @lhecker, but with a repo as popular as ours, I get something like 3,000 per day. So I can't quite tell which one you deleted, even with threaded mail. :(

Either way, I'm fine with this. I don't need to dig deeper.